jobmon_server/src/jobmon/server/web/models/workflow_run.py

```diff
@@ -120,6 +121,17 @@ def is_active(self) -> bool:
         """Statuses where Workflow Run is active (bound or running)."""
         return self.status in [WorkflowRunStatus.BOUND, WorkflowRunStatus.RUNNING]
 
+    def terminable(self, current_time: datetime.datetime) -> bool:
+        """Whether a workflowrun can be terminated.
+
+        A workflowrun can be terminated if it is in Cold/Hot resume state and has missed
+        the last reporting heartbeat.
+        """
+        return (
```

Comment:
I'd love for this method to be parameter-less but I can't find a way to get current time from the database without a session context. Let me know if you have any suggestions on how to do that. 

As far as I'm aware sqlalchemy function objects can't be evaluated outside a session, so I can't do something like `self.heartbeat_date <= func.now()`